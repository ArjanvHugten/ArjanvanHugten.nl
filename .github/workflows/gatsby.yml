name: Gatsby CI

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
    - name: Checkout
      uses: actions/checkout@v2.0.0
    - name: Build Gatsby Site
      uses: jzweifel/gatsby-cli-github-action@master
      with:
        gatsby-arg: build
    - name: Run Jest
      uses: stefanoeb/jest-action@1.0.3
      with:
        jestArgs: --silent
