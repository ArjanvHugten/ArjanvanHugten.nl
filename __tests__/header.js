import React from "react"
import renderer from "react-test-renderer"
import { render, screen } from '@testing-library/react'
import { useStaticQuery } from "gatsby"

import Header from "../src/components/layout/header"

beforeEach(() => {
    useStaticQuery.mockImplementationOnce(() => ({
      "navbarData": {
        "edges": [{
            "node": {
                "frontmatter": {
                    "menuItems": [{"label": "Home","linkURL": "/"},]
                }
            }
        }]
      }
    }))
})  

describe("Header", () => {
  it("Renders correctly", () => {
    const tree = renderer
      .create(<Header siteTitle="Default Starter" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Checks header text setted correctly", async () => {
    render(<Header siteTitle="Default Starter" />)

    const menuItems = await screen.queryAllByRole('link')

    expect(menuItems).toHaveLength(3)
    expect(screen.getByText("Default Starter")).toBeInTheDocument()
  })
})