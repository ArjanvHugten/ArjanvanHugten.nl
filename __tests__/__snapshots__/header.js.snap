// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders correctly 1`] = `
<header>
  <div
    className="container"
  >
    <nav
      aria-label="main navigation"
      className="navbar"
      role="navigation"
    >
      <div
        className="navbar-brand"
      >
        <a
          className="navbar-item primary-color"
          href="/"
        >
          Default Starter
        </a>
        <a
          aria-expanded="false"
          aria-label="menu"
          className="navbar-burger burger "
          data-target="navigationbar"
          href="#menu"
          onClick={[Function]}
          role="button"
        >
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
        </a>
      </div>
      <div
        className="navbar-menu "
        id="navigationbar"
      >
        <div
          className="navbar-start"
        >
          <a
            className="navbar-item"
            href="/"
          >
            Home
          </a>
        </div>
        <div
          className="navbar-end"
        >
          <div
            className="navbar-item"
          >
            <div
              className="buttons"
            >
              <a
                className="button is-primary"
                href="/contact-me/"
              >
                <strong>
                  Contact me
                </strong>
              </a>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </div>
</header>
`;
